src/App.tsx

```diff
@@ -15,36 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabetic = 1,
+  length = 2,
+}
+
+export const sortGoods = (goods, params) => {
+  const { sortBy, isReversed } = params;
+  const sortedGoods = [...goods];
+
+  if (sortBy) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.alphabetic:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          throw new Error('There is no such type of sort');
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(null);
+  const [isReversed, setReversed] = useState(false);
+
+  const sortedGoods = sortGoods(goodsFromServer, { sortBy, isReversed });
+
+  const resetAll = () => {
+    setSortBy(null);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortBy !== SortType.alphabetic,
+          })}
+          onClick={() => setSortBy(SortType.alphabetic)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortBy !== SortType.length,
+          })}
+          onClick={() => setSortBy(SortType.length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReversed(!isReversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortBy || isReversed) && (
```

Comment:
Create a variable for `sortBy || isReversed` to avoid a mix of different conditions